cocos/2d/renderer/render-data.ts

```diff
@@ -152,7 +153,7 @@ export class BaseRenderData {
     }
 
     // it should be invoked at where a render data is allocated.
-    public initRenderDrawInfo (comp: UIRenderer, drawInfoType: RenderDrawInfoType = RenderDrawInfoType.COMP) {
+    public initRenderDrawInfo (comp: UIRenderer | UIMeshRenderer, drawInfoType: RenderDrawInfoType = RenderDrawInfoType.COMP) {
```

Comment:
UIRenderer is the super class of UIMeshRenderer, why need to add ` | UIMeshRenderer` here?

Reply:
I've removed this union type, and added a TODO here
https://github.com/cocos/cocos-engine/pull/14572/commits/6cb79193a9d14dce3ce6fb9fa0412de9f8cd8033